File: php/flyweight/ChessFlyWeight.php
<?php

/**
 * 享元类
 *
 * 围棋 棋子颜色
 */
interface ChessFlyWeight
{
    public function setColor($color);
    public function getColor();

    public function display($coordinate);
}

/**
 * 具体享元类
 */
class ConcreteChess implements ChessFlyWeight
{
    private $color;

    public function __construct($color)
    {
        $this->color = $color;
    }
    public function setColor($color)
    {
        $this->color = $color;
    }

    public function getColor()
    {
        return $this->color;
    }

    public function display($coordinate)
    {
        echo "棋子颜色：", $this->color, PHP_EOL;
        echo "棋子位置：", $coordinate->getX(), "---", $coordinate->getY();
    }
}

File: java/pattern/src/flyweight/ChessFlyWeight.java
package flyweight;

/**
 * 享元类
 *
 * 围棋 棋子颜色
 */
public interface ChessFlyWeight {
    void setColor(String color);
    String getColor();
    void display(Coordinate c);
}

/**
 * 具体享元类
 */
class ConcreteChess implements ChessFlyWeight {
    private String color;

    public ConcreteChess(String color) {
        this.color = color;
    }

    @Override
    public void setColor(String color) {
        this.color = color;
    }

    @Override
    public String getColor() {
        return color;
    }

    @Override
    public void display(Coordinate c) {
        System.out.println("棋子颜色：" + color);
        System.out.println("棋子位置：" + c.getX() + "---" + c.getY());
    }
}